src/sentry/api/endpoints/dif_files.py

```diff
@@ -7,30 +7,31 @@
 
 from sentry.utils import json
 from sentry.api.serializers import serialize
-from sentry.api.bases.chunk import ChunkAssembleMixin
 from sentry.api.bases.project import ProjectEndpoint, ProjectReleasePermission
-from sentry.models import File, ChunkFileState, ProjectDSymFile
-
-
-class DifAssembleEndpoint(ChunkAssembleMixin, ProjectEndpoint):
+from sentry.models import ChunkFileState, ProjectDSymFile, FileBlob, \
+    FileBlobOwner, get_assemble_status, set_assemble_status
+
+
+def find_missing_chunks(organization, chunks):
+    """Returns a list of chunks which are missing for an org."""
+    if not chunks:
+        return []
+    blobs = list(FileBlob.objects.filter(checksum__in=chunks))
+    missing = set(chunks)
+    if not blobs:
+        return list(missing)
+    owned = set(x.blob_id for x in FileBlobOwner.objects.filter(
+        blob__in=[x.id for x in blobs]))
+    for blob in blobs:
```

Comment:
nit: last nit here, and this all can just be done with `set` operations instead.

```python
blob_ids = set(FileBlob.objects.filter(
    checksum__in=chunks,
).values_list('id', flat=True))
owned_ids = set(FileBlobOwner.objects.filter(
    blob_id__in=blob_ids,
).values_list('id', flat=True))
...
return list(blob_ids - owned_ids)
```

Reply:
Except we need to return checksums and not IDs...

Reply:
Gotcha. These nits aren't blocking anything fwiw. :)

Reply:
Err, and we should check that the FileBlobOwner's organization matches the parameter, too